src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -10,14 +10,14 @@
 public class Appointment {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Appointment should be in the format YYYY-MM-DD HH:MM HH:MM, and it should not be blank";
+            "Appointment should be in the format YYYY-MM-DD HH:MM HH:MM";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "\\d{4}-(0?[1-9]|1[0-2])-(0?[1-9]|[1-2]\\d|3[0-1])"
-            + " (2[0-3]:[0-5]?\\d|[0-1]?\\d:[0-5]?\\d) (2[0-3]:[0-5]?\\d|[0-1]?\\d:[0-5]?\\d)";
+    public static final String VALIDATION_REGEX = "(\\d{4}-(0?[1-9]|1[0-2])-(0?[1-9]|[1-2]\\d|3[0-1])"
+            + " (2[0-3]:[0-5]?\\d|[0-1]?\\d:[0-5]?\\d) (2[0-3]:[0-5]?\\d|[0-1]?\\d:[0-5]?\\d))|";
```

Comment:
Is this loosening of constraints necessary? If the user specifies the `ap/` field, they are likely trying to add an Appointment, so letting users be able to pass in an empty Appointment field does not make sense from a design standpoint. Additionally, the `t/` and `m/` fields do not allow users to pass empty fields either, so there may be a lack of consistency there.